app/src/main/kotlin/com/saltedge/authenticator/features/main/MainActivityViewModel.kt

```diff
@@ -0,0 +1,186 @@
+    }
+
+    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
+    fun onLifeCyclePause() {
+        connectivityReceiver.unregister(appContext)
+    }
+
+    /**
+     * Handle Network connection changes
+     */
+    override fun onNetworkConnectionChanged(isConnected: Boolean) {
+        internetConnectionWarningVisibility.postValue(if (isConnected) View.GONE else View.VISIBLE)
+    }
+
+    /**
+     * Handle Intents from external sources
+     */
+    fun onNewIntent(intent: Intent?) {
+        when {
+            intent.hasPendingAuthorizationData -> {
+                onShowAuthorizationDetailsEvent.postValue(
+                    ViewModelEvent(AuthorizationIdentifier(
+                        authorizationID = intent.authorizationId,
+                        connectionID = intent.connectionId
+                    ))
+                )
+            }
+            intent.hasDeepLinkData -> {
+                intent.deepLink.extractConnectAppLinkData()?.let { connectionAppLinkData ->
+                    onShowConnectEvent.postValue(ViewModelEvent(connectionAppLinkData))
+                } ?: intent.deepLink.extractActionAppLinkData()?.let { actionAppLinkData ->
+                    onShowSubmitActionEvent.postValue(ViewModelEvent(actionAppLinkData))
+                }
+            }
+        }
+    }
+
+    /**
+     * Handle result from QR Scanner
+     * if result is correct then show Connect View
+     */
+    fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        if (requestCode == QR_SCAN_REQUEST_CODE && resultCode == Activity.RESULT_OK) {
+            data?.let { onNewIntent(intent = it) }
+        }
+    }
+
+    /**
+     * Handle click on views
+     */
+    fun onViewClick(viewId: Int) {
+        when (viewId) {
+            R.id.appBarActionQrCode -> {
+                onQrScanClickEvent.postValue(ViewModelEvent(Unit))
+            }
+            R.id.appBarActionMenu -> {
+                onAppBarMenuClickEvent.postValue(ViewModelEvent(Unit))
+            }
+            R.id.appBarBackAction -> {
+                onAppBarMenuClickEvent.postValue(ViewModelEvent(Unit))
+            }
+        }
+    }
+
+    /**
+     * Handle new authorization event (e.g. from ActionSubmit)
+     */
+    override fun onNewAuthorization(authorizationIdentifier: AuthorizationIdentifier) {
+        onShowAuthorizationDetailsEvent.postValue(ViewModelEvent(authorizationIdentifier))
+    }
+
+    /**
+     * Handle app bar events
+     */
+    override fun updateAppbar(
+        titleResId: ResId?,
+        title: String?,
+        backActionImageResId: ResId?,
+        showMenu: Boolean
+    ) {
+        titleResId?.let { appBarTitle.postValue(appContext.getString(it)) }
+            ?: appBarTitle.postValue(title ?: "")
+        backActionImageResId?.let { appBarBackActionImage.postValue(backActionImageResId) }
+        appBarBackActionVisibility.postValue(if (backActionImageResId == null) View.GONE else View.VISIBLE)
```

Comment:
We have `setVisible` function, please use it.

Reply:
appBarBackActionVisibility - it is the event not view. we have discussed before about passing through event of boolean values